packages/mask/src/plugins/RealityCards/hooks/useDepositCallback.ts

```diff
@@ -0,0 +1,77 @@
+import { useCallback } from 'react'
+import BigNumber from 'bignumber.js'
+import { useAccount, useTransactionState, TransactionStateType, TransactionEventType } from '@masknet/web3-shared-evm'
+import { useTreasuryContract } from '../contracts/useTreasuryContract'
+
+/**
+ * A callback for deposit into reality cards
+ * @param amount
+ */
+export function useDepositCallback(amount: string) {
+    const account = useAccount()
+    const contract = useTreasuryContract()
+    const [state, setState] = useTransactionState()
+
+    const callback = useCallback(async () => {
+        if (!contract) {
+            setState({
+                type: TransactionStateType.UNKNOWN,
+            })
+            return
+        }
+
+        // pre-step: start waiting for provider to confirm tx
+        setState({
+            type: TransactionStateType.WAIT_FOR_CONFIRMING,
+        })
+
+        // step 1: estimate gas
+        const config = {
+            from: account,
+            value: new BigNumber(0).toFixed(),
+        }
+        const estimatedGas = await contract.methods
+            .deposit(amount, account)
+            .estimateGas(config)
+            .catch((error) => {
+                setState({
+                    type: TransactionStateType.FAILED,
+                    error,
+                })
+                throw error
+            })
+
+        // step 2: blocking
+        return new Promise<string>((resolve, reject) => {
+            contract.methods
+                .deposit(amount, account)
+                .send({
+                    ...config,
+                    gas: estimatedGas,
+                })
+                .on(TransactionEventType.CONFIRMATION, (confNumber, receipt) => {
```

Comment:
Does it necessary to wait for tx confirmation? Or TX_HASH is enough.

Reply:
I choose `tx confirmation` instead of `TX_HASH` because the next step depends on the deposit to be confirmed.

Technically it is possible to use `TX_HASH` but don't know what will happen to UX.